q2cli/core/usage.py

```diff
@@ -0,0 +1,231 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import collections
+import textwrap
+
+from qiime2.sdk import usage, util
+from qiime2.sdk.util import (
+    is_metadata_type,
+    is_visualization_type,
+)
+
+from q2cli.util import to_cli_name
+
+
+def is_iterable(val):
+    return isinstance(val, collections.abc.Iterable)
```

Comment:
Love this!  Didn't research this too deeply, but I never found a satisfying answer for reliably performing this check.